### DISCUSSION TITLE
Texture Format RGBA32Float Does Not Support Storage Texture Access StorageTextureAccess::ReadWrite

### DISCUSSION THREAD

**Discussion by Rakeshvcr** [gpuweb/gpuweb#4651] [Q&A] [answered]
I am using "wgpu" to create a render pipeline that supports both native and web environments. In my application, I utilize storage textures. The application runs fine natively, but on the web, I encounter the following issue:
Texture Format RGBA32Float Does Not Support Storage Texture Access StorageTextureAccess::ReadWrite
I have also tried changing the texture format to RGBA16Float, but the issue persists.

Chrome 124's WebGPU supports read_write access for storage textures, so this issue is unexpected.
Is there any known solution or workaround for this issue? Any help or guidance would be greatly appreciated.

Thank you!

**Comment by Jiawei-Shao** [comment, marked as answer]
Currently only `r32sint`, `r32uint` and `r32float` are supported as `read-write` storage binding (see the link [here](https://gpuweb.github.io/gpuweb/#plain-color-formats) (in `"read-write" STORAGE_BINDING` column). We need other WebGPU extensions to support `read-write` storage binding on more texture formats.
